geopandas/geodataframe.py

```diff
@@ -672,6 +672,25 @@ def explode(self):
         geo_df = df.set_geometry(self._geometry_column_name)
         return geo_df
 
+    #overrides the pandas.core.generic.NDFrame method
+    def astype(self, dtype, copy=True, errors='raise', **kwargs):
+        """
+        Temporary overwrite of pandas native astype method. Returns a
+        GeoDataFrame when geometries are valid and a pandas DataFrame when
+        the geometries are invalid.
+
+        Returns
+        -------
+        GeoDataFrame or DataFrame
+        """
+        df = super(DataFrame, self).astype(dtype, copy=copy, errors=errors, **kwargs)
+
+        try:
+            df = geopandas.GeoDataFrame(df, geometry=self._geometry_column_name)
+        except:
+            df = super(DataFrame, self).astype(dtype, copy=copy, errors=errors, **kwargs)
```

Comment:
You don't have to repeat the astype call here, you already did that above. So you can do here `return df` instead